source/distributed/tria.cc

```diff
@@ -3506,7 +3506,10 @@ namespace parallel
             cell->clear_refine_flag ();
             cell->clear_coarsen_flag ();
           }
-
+      this->prepare_coarsening_and_refinement ();
+      // If flags come out after mesh smoothing on ghost and artificial cells,
+      // we should keep them because they are resulted from smoothing of locally
+      // owned active cells. So no cleaning is needed here.
```

Comment:
There is no harm in that, but also no benefit because we only pass refinement flags along to p4est on locally owned cells. This is why I wrote the code in the first place that clears these flags just as a passive safety device: if we don't need these flags, then just remove them to make sure no downstream piece of code ever accidentally gets to see them.